state-synchronizer/src/coordinator.rs

```diff
@@ -513,15 +512,29 @@ impl<T: ExecutorProxyTrait> SyncCoordinator<T> {
         self.process_commit().await
     }
 
-    async fn validate_and_store_chunk(
-        &self,
+    fn validate_and_store_chunk(
+        &mut self,
         txn_list_with_proof: TransactionListWithProof,
         target: LedgerInfoWithSignatures,
     ) -> Result<()> {
+        let target_epoch_and_round = (target.ledger_info().epoch(), target.ledger_info().round());
+        let local_epoch_and_round = (
+            self.local_state.highest_local_li.ledger_info().epoch(),
+            self.local_state.highest_local_li.ledger_info().round(),
+        );
+        if target_epoch_and_round < local_epoch_and_round {
+            warn!(
+                "Ledger info is too old: local epoch/round: {:?}, epoch/round in request: {:?}.",
+                local_epoch_and_round, target_epoch_and_round,
+            );
+            return Ok(());
+        }
```

Comment:
this doesn't look right, why are we returning Ok(()) when this likely should be an error... even if we don't do anything with it.

Reply:
I wrote this piece of code - because to start a epoch, consensus needs to sync to a given point, but it may already at this point. As consensus there's no easy way to tell this, so I'd like to delegate this check to state sync(if it's already committed just return Ok).